server/render/index.js

```diff
@@ -78,13 +70,23 @@ export function serverRender( req, res ) {
 		context.i18nLocaleScript = '//widgets.wp.com/languages/calypso/' + context.lang + '.js';
 	}
 
-	if ( config.isEnabled( 'server-side-rendering' ) &&
-		context.store &&
-		context.layout &&
-		! context.user ) {
-		context.initialReduxState = pick( context.store.getState(), 'ui', 'themes' );
-		const key = context.renderCacheKey || JSON.stringify( context.layout );
-		Object.assign( context, render( context.layout, key ) );
+	if ( config.isEnabled( 'server-side-rendering' ) && ! context.user ) {
+		if ( context.layout ) {
+			const key = context.renderCacheKey || JSON.stringify( context.layout );
+			Object.assign( context, render( context.layout, key ) );
+		}
+
+		let title, metas = [], links = [];
+
+		if ( context.store ) {
+			title = getDocumentHeadFormattedTitle( context.store.getState() );
+			metas = getDocumentHeadMeta( context.store.getState() );
+			links = getDocumentHeadLink( context.store.getState() );
+
+			context.initialReduxState = pick( context.store.getState(), 'documentHead', 'ui', 'themes' );
+		}
+
+		context.head = { title, metas, links };
```

Comment:
Good call. This might've gone unnoticed in testing since the feature flag is on in all environments, and  `context.user` is notoriously unset on the server in `development`.